com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviourUpdater.cs

```diff
@@ -10,23 +11,185 @@ public class NetworkBehaviourUpdater
+        /// Under certain conditions, like user script invoking <see cref="NetworkVariableBase.SetUpdateTraits(NetworkVariableUpdateTraits)"/>
+        /// to define <see cref="NetworkVariableUpdateTraits"/>, a <see cref="NetworkObject"/> can remain in the
+        /// <see cref="m_DirtyNetworkObjects"/> list until the configured traits' conditions have been met.
+        /// </summary>
         private HashSet<NetworkObject> m_DirtyNetworkObjects = new HashSet<NetworkObject>();
+
+        /// <summary>
+        /// Contains any dirty <see cref="NetworkObject"/>s that will be added to the <see cref="m_DirtyNetworkObjects"/>
+        /// list on the next network tick (<see cref="OnNetworkTick"/>).
+        /// </summary>
         private HashSet<NetworkObject> m_PendingDirtyNetworkObjects = new HashSet<NetworkObject>();
 
 #if DEVELOPMENT_BUILD || UNITY_EDITOR
         private ProfilerMarker m_NetworkBehaviourUpdate = new ProfilerMarker($"{nameof(NetworkBehaviour)}.{nameof(NetworkBehaviourUpdate)}");
 #endif
 
+        /// <summary>
+        /// Adds a <see cref="NetworkObject"/> to the prending dirty list.
+        /// The <see cref="m_PendingDirtyNetworkObjects"/> list is merged into the <see cref="m_DirtyNetworkObjects"/> list
+        /// when processed.
+        /// </summary>
         internal void AddForUpdate(NetworkObject networkObject)
         {
             // Since this is a HashSet, we don't need to worry about duplicate entries
             m_PendingDirtyNetworkObjects.Add(networkObject);
         }
 
+        /// <summary>
+        /// (Client-server network topology only)
+        /// The server handles processing network variables the same way as a client
+        /// with the primary difference being that the server sends updates to all
+        /// observers.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void ProcessDirtyObjectServer(NetworkObject dirtyObj, bool forceSend)
+        {
+            for (int i = 0; i < m_ConnectionManager.ConnectedClientsList.Count; i++)
+            {
+                var client = m_ConnectionManager.ConnectedClientsList[i];
+                if (m_NetworkManager.DistributedAuthorityMode || dirtyObj.IsNetworkVisibleTo(client.ClientId))
+                {
+                    // Sync just the variables for just the objects this client sees
+                    for (int k = 0; k < dirtyObj.ChildNetworkBehaviours.Count; k++)
+                    {
+                        dirtyObj.ChildNetworkBehaviours[k].NetworkVariableUpdate(client.ClientId, forceSend);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Clients handle processing dirty objects relative to the client.
+        /// The <see cref="NetworkVariableDeltaMessage"/> is client to server.
+        /// With distributed authority live service sessions, this is sent to
+        /// the Rust server.
```

Comment:
nit: we kind of avoid directly mentioning the language of the CMB service in the codebase

```suggestion
        /// the CMB service.
```

p.s I love this change! ❤️